pkg/ccl/backupccl/backup_test.go

```diff
@@ -8703,3 +8703,25 @@ DROP TABLE foo;
 	defer cleanupEmptyCluster()
 	sqlDBRestore.Exec(t, "RESTORE FROM $1 AS OF SYSTEM TIME "+aost, LocalFoo)
 }
+
+func TestRestoreNewDatabaseName(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	const numAccounts = 1
+	_, _, sqlDB, _, cleanupFn := BackupRestoreTestSetup(t, singleNode, numAccounts, InitManualReplication)
+	defer cleanupFn()
+
+	sqlDB.Exec(t, `BACKUP DATABASE data TO $1`, LocalFoo)
+
+	// Should fail because 'data' database is still in cluster
+	sqlDB.ExpectErr(t, "database",
```

Comment:
we should match more of the error, just matching "database" doesn't seem like a strong enough assertion.

Reply:
done